chapters/categories.tex

```diff
@@ -299,29 +299,122 @@ \section{Monos and epis}
+  \begin{eqnarray*}
+  0 [ \vec{x},x ] & = & x \\ 
+  (1+i) [ \vec{x},x ]& = & i [\vec{x} ]\\
+  (t\,u) [\vec{x}] & = & (t [\vec{x}]) \,(u[\vec{x}) \\
+  (\lambda^\sigma t) [\vec{x}] & = & \lambda (t [\vec{x} \rhd \sigma])    
+  \end{eqnarray*}
+We now define identity : Given $\Gamma = \sigma_n . \sigma_{n-1} \dots .\sigma_0$ we define
+\begin{align*}
+  & \id_\Gamma  :   \Subst(\Gamma,\Gamma) \\
+  & \id_\Gamma = n , n-1 , \dots, 0 
+\end{align*}
+% Actually, we observe that there is a nice recursive definition of $\id$:
+% \begin{align*}
+% & \id_\bullet = \epsilon \\
+% & \id_{\Gamma\rhd\sigma} = \id_\Gamma \rhd \sigma 
+% \end{align*}
+
+Using renaming we can now define $1+$ for terms, that is given $\Tm(\Gamma,\sigma)$ we define 
+\begin{align*}
+1+t & : \Tm(\Gamma\rhd \sigma,\tau) \\
+1+t & = t [1+\id]
+\end{align*}
+% Repeating the definition for renamings we can define operations on substitutions:
+% Given $\vec{t} = t_n,t_{n-1},\dots,t_0: \Subst(\Gamma,\Delta) $, we can apply $1+$ to it componentwise:
+% \begin{align*}
+%   & 1 + \vec{t} : \Subst (\Gamma\rhd \sigma,\Delta) \\
+%   & 1 + \vec{t} = 1+t_n , 1+t_{n-1} , \dots , 1+t_0\\
+%   \\
+%   & \vec{x} \rhd \sigma  : \Vars(\Gamma \rhd\sigma,\Delta\rhd\sigma)\\
+%   & \vec{x} \rhd \sigma  = (1 + \vec{x},  0)
+% \end{align*}
+
+% We first define renaming, that is $t : \Tm(\Gamma,\sigma)$ and 
+% $\vec{x} : \Vars(\Delta,\Gamma)$ we define $t[\vec{x}] : \Tm(\Delta,\sigma)$ by the following equations:
+%   \begin{eqnarray*}
+%   0 [ \vec{x},x ] & = & x \\ 
+%   (1+i) [ \vec{x},x ]& = & x [\vec{x} ]\\
+%   (t\,u) [\vec{x}] & = & (t [\vec{x}]) \,(u[\vec{x}) \\
+%   (\lambda^\sigma t) [\vec{x}] & = & \lambda (t [\vec{x} \rhd \sigma])    
+%   \end{eqnarray*}
+
+We can now repeat the definitions for renamings to define the corresponding operations on terms, i.e. given $\vec{u}: \Subst(\Gamma,\Delta)$ and $t : \Tm(\Delta,\sigma)$ we define :
+\begin{align*}
+  & \vec{u} \rhd \sigma  : \Subst(\Gamma \rhd\sigma,\Delta\rhd\sigma)  \\
+  & t[\vec{u}] : \Tm(\Gamma,\sigma) 
+\end{align*}
+It seems to be quite inelegant to repeat the same definition but it isn't so easy to avoid this, since we need renamings to derive the $1+$-operation for terms. There are alternatives but each comes with their own issues.
+
+We can now define composition: given $\vec{t} = t_n,t_{n-1},\dots, t_0 : \Subst(\Gamma,\Delta)$ and $\vec{u} : \Subst(\Theta,\Gamma)$ we define
+\begin{align*}
+  & \vec{t} \circ \vec{u} : \Subst(\Theta,\Delta) \\
+  & \vec{t} \circ \vec{u}  = t_n[\vec{u}] , t_{n-1}[\vec{u}], \dots t_0[\vec{u}]
+\end{align*}
+
+  \begin{exercise}
+    Show that lambda terms and substitutions form a category $\cat{\Lambda}$ where the objects are contexts, given contexts $\Gamma,\Delta$ the homsets are substitutions $\Subst(\Gamma,\Delta)$ and identity and composition are defined as above. You need to check the laws.
```

Comment:
This info should've been presented at the beginning. Maybe split the different things to check into different exercises (e.g. one exercise is to verify associativity, one to check unit laws, etc.), and then conclude at the end of the section that it's a category?